iotdb/src/main/java/org/apache/iotdb/db/engine/modification/io/LocalTextModificationAccessor.java

```diff
@@ -0,0 +1,138 @@
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.engine.modification.io;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+
+import org.apache.iotdb.db.engine.modification.Deletion;
+import org.apache.iotdb.db.engine.modification.Modification;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * LocalTextModificationAccessor uses a file on local file system to store the modifications
+ * in text format, and writes modifications by appending to the tail of the file.
+ */
+public class LocalTextModificationAccessor implements ModificationReader, ModificationWriter {
+
+  private static final Logger logger = LoggerFactory.getLogger(LocalTextModificationAccessor.class);
+  private static final String SEPARATOR = ",";
+
+  private String filePath;
+  private BufferedWriter writer;
+
+  /**
+   * Construct a LocalTextModificationAccessor using a file specified by filePath. Only a writer
+   * will be created because the reader will be created only if necessary(call of read()).
+   *
+   * @param filePath the path of the file that is used for storing modifications.
+   * @throws IOException if the writer cannot be created.
+   */
+  public LocalTextModificationAccessor(String filePath) throws IOException {
+    this.filePath = filePath;
+    writer = new BufferedWriter(new FileWriter(filePath));
+  }
+
+  @Override
+  public Collection<Modification> read() throws IOException {
+    BufferedReader reader;
+    try {
+      reader = new BufferedReader(new FileReader(filePath));
+    } catch (FileNotFoundException e) {
+      return null;
+    }
+    String line;
+
+    List<Modification> modificationList = new ArrayList<>();
+    try {
+      while ((line = reader.readLine()) != null) {
+        modificationList.add(decodeModification(line));
+      }
+    } catch (IOException e) {
+      reader.close();
+      logger.error("An error occurred when reading modifications, and the remaining modifications "
+              + "were ignored.", e);
+    }
+    return modificationList;
+  }
+
+  @Override
+  public void close() throws IOException {
+    writer.close();
+  }
+
+  @Override
+  public void write(Modification mod) throws IOException {
+    writer.write(encodeModification(mod));
+    writer.newLine();
+    writer.flush();
+  }
+
+  private static String encodeModification(Modification mod) {
+    if (mod instanceof Deletion)
+      return encodeDeletion((Deletion) mod);
+    return null;
+  }
+
+  private static Modification decodeModification(String src) throws IOException {
+    String[] fields = src.split(SEPARATOR);
+    if (Modification.Type.DELETION.name().equals(fields[0])) {
```

Comment:
Using name() is wasteful.  how about using the ordinal() of the Enum.

Reply:
Using ordinal will introduce a translation from string to integer in decoding and a translation from integer to string in encoding, so this will be a trade off between space and time.
If you still prefer space more, I will do the change. However, adding a binary form of Modification in the future will be more space-efficient.